src/Namotion.Reflection.Tests/GenericsTests.cs

```diff
@@ -0,0 +1,116 @@
+            {
+                Assert.Equal(expectedNullability, t.GetTypeInfo().GenericTypeParameters[0].ToContextualType().Nullability);
+                Assert.Equal(expectedNullability, t.GetProperty("Prop")!.ToContextualProperty().Nullability);
+                var method = t.GetMethod("DoStuff")!;
+                Assert.Equal(expectedNullability, method.ReturnParameter.ToContextualParameter().Nullability);
+                Assert.Equal(expectedNullability, method.GetParameters()[0].ToContextualParameter().Nullability);
+            }
+
+            DoTest(typeof(UnconstrainedGenericClass<>), Nullability.Nullable);
+            DoTest(typeof(NotNullGenericClass<>), Nullability.NotNullable);
+            DoTest(typeof(NullableGenericClass<>), Nullability.Nullable);
+            DoTest(typeof(StructGenericClass<>), Nullability.NotNullable);
+        }
+
+        public class ClosedGenericsClass
+        {
+            public UnconstrainedGenericClass<string> NotNull1 { get; set; }
+            public NotNullGenericClass<string> NotNull2 { get; set; }
+            public NullableGenericClass<string> NotNull3 { get; set; }
+
+            public StructGenericClass<int> Struct { get; set; }
+
+
+            public UnconstrainedGenericClass<string?> Nullable1 { get; set; }
+            public NullableGenericClass<string?> Nullable2 { get; set; }
+        }
+
+        [Fact]
+        public void ClosedGenericsType()
+        {
+            void DoTest(string propertyName, Nullability expectedNullability)
+            {
+                var property = typeof(ClosedGenericsClass).GetProperty(propertyName)!.ToContextualProperty();
+                Assert.Equal(expectedNullability, property.GenericArguments[0].Nullability);
+                Assert.Equal(expectedNullability, property.GetProperty("Prop")!.Nullability);// TODO : fails...
+            }
+
+            /*DoTest(nameof(ClosedGenericsClass.NotNull1), Nullability.NotNullable);
+            DoTest(nameof(ClosedGenericsClass.NotNull2), Nullability.NotNullable);
+            DoTest(nameof(ClosedGenericsClass.NotNull3), Nullability.NotNullable);
+            DoTest(nameof(ClosedGenericsClass.Struct), Nullability.NotNullable);*/
+
+            DoTest(nameof(ClosedGenericsClass.Nullable1), Nullability.Nullable);
+            // DoTest(nameof(ClosedGenericsClass.Nullable2), Nullability.Nullable);
+        }
+
+        public class GenericClass<T>
+        {
+            public T GetT()
+            {
+                return default!;
+            }
+
+            public void SetT(T value)
+            {
+            }
+
+            public T Prop { get; set; } = default!;
+        }
+
+        public class GenericClassContainer
+        {
+            public GenericClass<string?> Nullable = new GenericClass<string?>();
+            public GenericClass<string> NonNullable = new GenericClass<string>();
+        }
+
+        [Fact]
+        public void GenericClassResolutionTest()
```

Comment:
I don't get how this test is different from the ClosedGenericsType and the ClosedGenericsClass declared above. Is that just a test on a field rather than a property ?

Reply:
I dont understand your question.

Reply:
You introduced this test in a commit after mine, but  I don't get how it is diffrent from these tests: https://github.com/RicoSuter/Namotion.Reflection/pull/60/files#diff-82e5a6a45918474d02929539d6aa5f4158e8b2f81f985f186b911c8971ae7baaR77

Reply:
Hmm? No this is your test, no? 🙂

Reply:
I only added the field asserts to also cover this

Reply:
This is yours : https://github.com/RicoSuter/Namotion.Reflection/pull/60/commits/b4a4161e320f9eb1cb0b7ea37ded5eebc98f99e4

Reply:
Removed.